src/waveform/renderers/waveformmark.cpp

```diff
@@ -93,6 +93,75 @@ bool isShowUntilNextPositionControl(const QString& positionControl) {
 
 } // anonymous namespace
 
+WaveformMark::WaveformMark(const QString& group,
+        QString positionControl,
+        const QString& visibilityControl,
+        const QString& textColor,
+        const QString& markAlign,
+        const QString& text,
+        const QString& pixmapPath,
+        const QString& iconPath,
+        QColor color,
+        int priority,
+        int hotCue,
+        const WaveformSignalColors& signalColors)
+        : m_textColor(textColor),
+          m_pixmapPath(pixmapPath),
+          m_iconPath(iconPath),
+          m_linePosition{},
+          m_breadth{},
+          m_level{},
+          m_iPriority(priority),
+          m_iHotCue(hotCue),
+          m_showUntilNext{} {
+    QString endPositionControl;
+    QString typeControl;
+    if (hotCue != Cue::kNoHotCue) {
+        positionControl = "hotcue_" + QString::number(hotCue + 1) + "_position";
+        endPositionControl = "hotcue_" + QString::number(hotCue + 1) + "_endposition";
+        typeControl = "hotcue_" + QString::number(hotCue + 1) + "_type";
```

Comment:
NIT: save the `QString::number` call in a local var. Seems unnecessary to call this often.